### PR TITLE
Wst 180 blured input 

### PR DESCRIPTION
**Pull request by cnolimit**

### CODE REVIEW

File: src/components/InputMoney.js [hunk 1]
```diff
@@ -64,83 +64,70 @@ class InputMoney extends Component {
...
+    typeof this.props.onChange === "function" && this.props.onChange(formatedValue);
     if (value.length) this.setState({ hasValue: true });
     else this.setState({ hasValue: false });
     return formatedValue;
   };
 
-  _onBlur = value => {
+  _onBlur = elm => {
+    let valueObj = elm;
+    if (!elm.value.length) this.setState({ hasValue: false });
```
Comment (IljaDaderko) [inline src/components/InputMoney.js]: There was similar func, that also included check `if (!elm) return null;` is it missing here?

File: src/components/InputPassword.js [hunk 2]
```diff
@@ -9,6 +9,8 @@ const zoomFade = keyframes`
   100% { transform: scale(2); opacity: 0; }
 `;
 
+const SInput = styled(Input)`width: 100%;`;
```
Comment (IljaDaderko) [inline src/components/InputPassword.js]: These keep popping up, can we set `width: 100%` on `input` itself so we don't need to use `SInput`'s ?

File: src/components/InputPassword.js [hunk 3]
```diff
@@ -83,10 +87,6 @@ class InputPassword extends Component {
     password: ""
   };
 
-  componentDidMount() {
```
Comment (IljaDaderko) [inline src/components/InputPassword.js]: Are you sure they need to be removed? i.e. Component should return ref for input when it mounts so parent application has access to it immediately.

Comment (cnolimit) [reply src/components/InputPassword.js]: No, okay so whats happening is `<InputPassword />` extends `<Input />` which will handle the getRef as follows ...
screen shot 2017-10-04 at 10 38 32

screen shot 2017-10-04 at 10 36 34

`getRef` is handled by `<Input />` so it is available as soon as the component mounts, which is why components which extend it don't need to implement the same logic, the prop is hoisted.

Comment (cnolimit) [reply src/components/InputPassword.js]: Any component which doesn't implement it i.e `<InputIncrement />` will implement the logic in there scope...
screen shot 2017-10-04 at 10 43 04

File: src/components/InputMoney.js (hunk 1 above)
Comment (cnolimit) [reply src/components/InputMoney.js]: tbh the code `if (!elm) return null;` is actually redundant as the a value will always exits coming into the `_onBlur` function either natively or manually

Comment (cnolimit) [reply src/components/InputMoney.js]: the purpose of the code was to set the `valueObj = elm` and have it go through validation check and mutate `valueObj` accordingly else return its initial assigned value

File: src/components/InputPassword.js (hunk 2 above)
Comment (cnolimit) [reply src/components/InputPassword.js]: yeah i was having issues without it but i've just solved it, i can't find any other places where it exist though did i miss any?

**Pull request merged**